server/tmserver/world.py

```diff
@@ -34,6 +34,8 @@ def dispatch_action(cls, sender_obj, action, action_args):
             cls.handle_whisper(sender_obj, action_args)
         if action == 'look':
             cls.handle_look(sender_obj, action_args)
+        if action == 'refresh':
```

Comment:
would a user ever be typing `/refresh`? i'm thinking the client will be performing this action at some polling interval, sending a message like `DATA roominfo` and getting back a payload like `DATA roominfo {...}`. Essentially, we're crudely reinventing REST for use in a bidirectional websocket connection (instead of a unidirectional HTTP setup).

As we need JSON payloads we can add handlers for other kinds of data -- like `minimap`, `map` (for maybe a big map modal), `inventory`, `topic`...

does this make sense? i can whip up the DATA handler on the server side today, i think.

Reply:
yeah, realistically the user would never actually need to manually `/refresh`, that's mostly a debugging/instant gratification thing at the moment. but, i wonder if instead of having the client regularly poll, we just have the server update the client whenever there's actually a relevant change (new object enters the room, or you move into a different room, or the room name/description gets changed, etc.), and the client refreshes the panels when it gets fresh DATA. it seems like less work on both ends (no need to transmit DATA payloads if they haven't changed, not missing information if something happens and then unhappens within the polling interval).